newIDE/app/src/ResourcesList/ResourceUtils.js

```diff
@@ -174,6 +174,66 @@ export const renameResourcesInProject = (
   resourcesRenamer.delete();
 };
 
+export const parseLocalFilePathOrExtensionFromMetadata = (
+  resource: gdResource
+): {|
+  localFilePath: ?string,
+  extension: ?string,
+|} => {
+  const metadataAsString = resource.getMetadata();
+  if (metadataAsString) {
+    try {
+      const metadata = JSON.parse(metadataAsString);
+      if (metadata && typeof metadata === 'object') {
+        return {
+          localFilePath:
+            metadata.localFilePath && typeof metadata.localFilePath === 'string'
+              ? metadata.localFilePath
+              : null,
+          extension:
+            metadata.extension && typeof metadata.extension === 'string'
+              ? metadata.extension
+              : null,
+        };
+      }
+    } catch (error) {
+      console.warn(
+        'Malformed metadata for resource with name ' +
+          resource.getName() +
+          ' - ignoring it.'
+      );
+    }
+  }
+
+  return {
+    localFilePath: null,
+    extension: null,
+  };
+};
+
+export const updateResourceJsonMetadata = (
+  resource: gdResource,
+  newMetadata: { [string]: any }
+) => {
+  const metadataAsString = resource.getMetadata();
+  try {
+    const existingMetadata = metadataAsString
+      ? JSON.parse(metadataAsString)
+      : {};
+    resource.setMetadata(
+      JSON.stringify({
+        ...existingMetadata,
+        ...newMetadata,
+      })
+    );
+    return;
+  } catch (error) {
+    // Ignore the error.
```

Comment:
why?

Reply:
If the metadata is not valid JSON, we'll just don't try to merge it with the new metadata, but we'll overwrite the metadata entirely. I updated the comment.